lib/manageiq/environment.rb

```diff
@@ -121,6 +121,46 @@ def self.bundler_version
       File.read(gemfile).match(/gem\s+['"]bundler['"],\s+['"](.+?)['"]/)[1]
     end
 
+    def self.warn_if_branch_out_of_date
+      return if %w(production test).include?(ENV['RAILS_ENV'].to_s)
+
+      Dir.chdir(APP_ROOT) do
+        system!("git fetch")
+
+        commits =
+          if upstream_tracking_branch.empty?
+            commits_behind_upstream
+          else
+            commits_behind_tracked_branch
```

Comment:
So this won't help in the case of a feature branch falling behind `upstream/master` right?  It would only say that `feature_branch` was behind `agrare/feature_branch` ?

Reply:
Yeah, if your feature branch is tracking your fork instead of upstream, you wouldn't see this information.  Do people do that?  Unless you're working on a long feature involving multiple developers, you should probably track upstream, right?  I can check the tracked branch AND upstream instead of OR upstream but I thought that was redundant for most people.

Reply:
Actually, I can just drop this whole else and `commits_behind_tracked_branch` if we don't care if you're tracking branches and we want to assume upstream/master.